enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,974 @@
+for the workflow for this user story.
+
+#### ROSA HCP: 1 MachinePool in 1 Availability Zone
+
+_"As a user of ROSA HCP, I want to install a cluster with 1 MachinePool in 1
+Availability Zone and have my `default` IngressController only mapped to the subnets
+in that Availability Zone."_
+
+Similar to [Day 1 Default IngressController Subnet Selection on AWS](#day-1-default-ingresscontroller-subnet-selection-on-aws).
+
+When a cluster is installed into a VPC containing other subnets without the
+`kubernetes.io/cluster/<cluster-id>` tag in different AZs than the cluster-role subnets,
+the `default` IngressController's load balancer may select those subnets.
+This creates a security risk because the load balancer may select subnets that belong to other clusters. This
+user story is derived from [OCM-5969](https://issues.redhat.com/browse/OCM-5969).
+
+See [Setting Default IngressController subnets during Installation Workflow](#setting-default-ingresscontroller-subnets-during-installation-workflow)
+for the workflow for this user story.
+
+#### ROSA HCP: Adding a 2nd MachinePool
+
+_"As a user of ROSA HCP, I want to install a cluster with 1 MachinePool in 1
+AZ, then add a 2nd MachinePool in another AZ and update my IngressController's
+subnets to map to the new AZ."_
+
+A user installed a ROSA HCP cluster with 1 MachinePool in one AZ similar to
+[ROSA HCP: 1 MachinePool in 1 Availability Zone](#rosa-hcp-1-machinepool-in-1-availability-zone),
+but now they want to add another machine pool in a different AZ. The user
+would need to adjust the existing default IngressController subnets by following
+the workflow [Updating an existing IngressController with new Subnets Workflow](/enhancements/ingress/lb-subnet-selection-aws.md#updating-an-existing-ingresscontroller-with-new-subnets-workflow).
+
+Additionally, if the user wants all future IngressControllers to use the new subnet by default, see
+[Updating the Default Subnets for New IngressControllers Workflow](#updating-the-default-subnets-for-new-ingresscontrollers-workflow).
+
+#### Dedicated IngressController Subnets for Security
+
+_"As a cluster admin, I want to install a cluster with all IngressControllers
+using a dedicated subnets in order to have load balancer VIPs on a dedicated VLAN for
+applying ACLs."_
+
+A cluster admin wants their all of their load balancers mapped to separate subnet(s), allowing
+firewall ACL rules to specifically target the load balancer VLAN
+(see https://access.redhat.com/support/cases/#/case/03054638). The dedicated subnet(s)
+must be located in the same AZ(s) as the provided cluster-role subnets; otherwise,
+the IngressController will not function properly.
+
+See [Setting Default IngressController subnets during Installation Workflow](#setting-default-ingresscontroller-subnets-during-installation-workflow)
+for the workflow for this user story.
+
```

Comment:
Wait, I just had an realization, we are creating a new "subnets" field, can't we disregard the need for compatibility?
 
By that, I mean, maybe we can solve [OCPBUGS-17432](https://issues.redhat.com/browse/OCPBUGS-17432) while we do this `subnets` field deprecation at the same time? Since the users have to explicitly opt into the new API, can't we implement a different behavior?

If we can, I think that will massively simplify this API and proposal, like @mtulio has suggested before and make these non-goals, now goals:
```
- Assume that bring-your-own (BYO) cluster-role subnets are also IngressController-role subnets.
- Assume that IngressController-role subnets are also cluster-role subnets.
```

If we assume Cluster Role Subnets == IngressController Role subnets, then:
- The API is simplified (no need for `role` fields, it's all the same)
- No need to explain the role concept in this proposal
- Customers won't being surprised when they install a cluster in an existing VPC and their `default` ingress controller selects subnets external to there cluster (OCPBUGS-17432)
- Customers won't be suprised when they install a private cluster in an existing VPC and their `default` ingress controller selects a public subnet [RFE-2816](https://issues.redhat.com/browse/RFE-2816) 
- We'd be effectively taking full control of the default subnet selection process for IngressControllers from what AWS decided in [kubernetes/kubernetes#97431](https://github.com/kubernetes/kubernetes/pull/97431)
- It's more obvious to customers because there is no ambiguity about which subnets are being used when installing a cluster, better UX

We don't have any users stories that where users actually **want** to use subnets outside of what they specified as cluster subnets (all users stories are the opposite...stop selecting external subnets). Should we leverage this deprecation of `subnets` to make thing work as expected? Any reason why I am missing something here? @JoelSpeed @mtulio @patrickdillon

Reply:
Sounds like a simpler approach to me if the user stories are talking about stopping using ingress subnets outside of the cluster subnets. Do we have any use cases where the ingress subsets are a subset of the cluster subnets, or is it always that they are a matching set?

Reply:
> Do we have any use cases where the ingress subsets are a subset of the cluster subnets, or is it always that they are a matching set?

I've been looking. I thought maybe AWS Outposts would potentially provide a use case for this, but I don't think so after some digging. For AWS Outpost, it seems Cluster-Role subnets still should == IngressController subnets (via https://github.com/openshift/enhancements/pull/1260). So please, let me know if anyone has any use cases of when IngressController subnets != Cluster-Role Subnets.

But not to say there is an edge case that might come up later. I think the important thing is that we design the API in such a way that it _could_ be extended to accommodate different subnet usages in the future.

Reply:
Just following up on this thread for completeness. I've since spoken to Service Delivery and they advocated for a dedicated IngressController subnet user story (which I removed, but now I will add back in a future PR push), which means we need a new API and `Cluster` & `IngressController` roles regardless.

We've since decided to support subnet discovery as a no-opinion path (no IngressController-role subnets specified). See meeting [docs](https://docs.google.com/document/d/1GOENio9rg3rj0lRn1Sz0DqguqLsNfhwr2GOjOiR6E6I/edit?tab=t.0) for more details on why.